File: src/PortfolioTracker.Ui/src/app/models/analyticsResponse.ts
export interface CurrencyAnalytic {
  currency: number;
  portfolioAmount: number;
  portfolioPercent: number;
}

export interface AccountAnalytic {
  accountId: string;
  accountName: string;
  accountType: number;
  portfolioAmount: number;
  portfolioPercent: number;
}

export interface AssetTypeAnalytic {
  assetType: number;
  portfolioAmount: number;
  portfolioPercent: number;
}

export interface RiskLevelAnalytic {
  riskLevel: number;
  portfolioAmount: number;
  portfolioPercent: number;
}

export interface Snapshot {
  id: string;
  userId: string;
  generationTime: Date;
  totalAmount: number;
  currencyAnalytics: CurrencyAnalytic[];
  accountAnalytics: AccountAnalytic[];
  assetTypeAnalytics: AssetTypeAnalytic[];
  riskLevelAnalytics: RiskLevelAnalytic[];
  transactionTypeAnalytics?: any;
}

export interface AnalyticsResponse {
  mostPopularCurrency: string;
  highestLevelOfRisk: string;
  numberOfAsstTypes: number;
  numberOfAccounts: number;
  currentTotalAmount: number;
  portfolioChange: number;
  snapshots: Snapshot[];
}

File: src/PortfolioTracker.Api/Dto/AnalyticsResponse.cs
using PortfolioTracker.DataAccess.Models;
using PortfolioTracker.Events.Common;

namespace PortfolioTracker.Api.Dto
{
    public class AnalyticsResponse
    {
        public string MostPopularCurrency { get; set; }
        public string HighestLevelOfRisk { get; set; }
        public int NumberOfAsstTypes { get; set; }
        public int NumberOfAccounts { get; set; }
        public decimal CurrentTotalAmount { get; set; }
        public decimal PortfolioChange { get; set; }
        public List<Snapshot> Snapshots { get; set; } = null!;
    }
}
